include/openrave/geometry.h

```diff
@@ -383,11 +383,15 @@ class RaveTransform
     template <typename U> RaveTransform(const RaveTransform<U>& t) {
         rot = t.rot;
         trans = t.trans;
-        MATH_ASSERT( rot.lengthsqr4() > 0.99f && rot.lengthsqr4() < 1.01f );
+        const T l = rot.lengthsqr4();
```

Comment:
i think this is a good catch. for further improvement, computation of `l` can be skipped when assert is disabled. by doing like
```c++
#ifdef BOOST_ASSERT
#define MATH_ASSERT BOOST_ASSERT
#ifdef BOOST_DISABLE_ASSERTS
#define MATH_DISABLE_ASSERTS
#endif 
#else
#include <cassert>
#define MATH_ASSERT assert
#ifdef NDEBUG
#define MATH_DISABLE_ASSERTS
#endif 
#endif

...
#ifndef MATH_DISABLE_ASSERTS
        const T l = rot.lengthsqr4();
        MATH_ASSERT( l > 0.99f && l < 1.01f );
#endif
```

Reply:
good idea. incorporated. many thanks commenting!